Source/BusinessLogic/Jobs/BoardGameGeekCleanUpService/BoardGameGeekCleanUpService.cs

```diff
@@ -0,0 +1,168 @@
+                                else
+                                {
+                                    var newRecord = CreateBGGGame(gameToAdd);
+
+                                    UpdateGameDefinition(game, newRecord.Id, result);
+                                }
+                            }
+                        }
+                    }
+
+                    if (game.BoardGameGeekGameDefinitionId != null)
+                    {
+                        _dataContext.CommitAllChanges();
+                    }
+                    else
+                    {
+                        result.StillOrphanGames.Add(new LinkOrphanGamesResult.OrphanGame()
+                        {
+                            Name = game.Name,
+                            Id = game.Id,
+                            GamingGroupId = game.GamingGroupId
+                        });
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                result.Success = false;
+                _rollbar.SendException(ex);
+            }
+
+            stopwatch.Stop();
+            result.TimeEllapsed = stopwatch.Elapsed;
+            return result;
+        }
+
+        private void UpdateGameDefinition(GameDefinition game, int boardGameGeekGameDefinitionId,
+            LinkOrphanGamesResult result)
+        {
+            game.BoardGameGeekGameDefinitionId = boardGameGeekGameDefinitionId;
+            result.LinkedGames++;
+        }
+
+        private BoardGameGeekGameDefinition CreateBGGGame(GameDetails gameToAdd)
+        {
+            var newRecord = new BoardGameGeekGameDefinition
+            {
+                Id = gameToAdd.GameId,
+                Name = gameToAdd.Name,
+                Thumbnail = gameToAdd.Thumbnail,
+                MaxPlayers = gameToAdd.MaxPlayers,
+                MinPlayers = gameToAdd.MinPlayers,
+                PlayingTime = gameToAdd.PlayingTime,
+                AverageWeight = gameToAdd.AverageWeight,
+                Description = gameToAdd.Description
+            };
+
+            _dataContext.Save(newRecord, new ApplicationUser());
+            return newRecord;
+        }
+
+        private BoardGameGeekGameDefinition GetExistingBGGGameById(GameDetails gameToAdd)
+        {
+            var existingGame = _dataContext.GetQueryable<BoardGameGeekGameDefinition>()
+                .FirstOrDefault(bgg => bgg.Id == gameToAdd.GameId);
+            return existingGame;
+        }
+
+        private GameDetails GetGameToAddFromSearch(List<SearchBoardGameResult> searchResult)
+        {
+            var gamesToAdd =
+                searchResult.Select(
+                    searchBoardGameResult => _boardGameGeekApiClient.GetGameDetails(searchBoardGameResult.BoardGameId))
+                    .Where(gameDetails => gameDetails != null)
+                    .ToList();
+
+            var gameToAdd =
+                gamesToAdd.Where(g => g.Thumbnail != null)
```

Comment:
This must be my next issue... sorry for the spam :_(

El dom., 13 mar. 2016 a las 18:29, Jake Gordon (<notifications@github.com>)
escribió:

> In
> Source/BusinessLogic/Jobs/BoardGameGeekCleanUpService/BoardGameGeekCleanUpService.cs
> <https://github.com/NemeStats/NemeStats/pull/377#discussion_r55940496>:
>
> > +        {
> > +            var existingGame = _dataContext.GetQueryable<BoardGameGeekGameDefinition>()
> > +                .FirstOrDefault(bgg => bgg.Id == gameToAdd.GameId);
> > +            return existingGame;
> > +        }
> > +
> > +        private GameDetails GetGameToAddFromSearch(List<SearchBoardGameResult> searchResult)
> > +        {
> > +            var gamesToAdd =
> > +                searchResult.Select(
> > +                    searchBoardGameResult => _boardGameGeekApiClient.GetGameDetails(searchBoardGameResult.BoardGameId))
> > +                    .Where(gameDetails => gameDetails != null)
> > +                    .ToList();
> > +
> > +            var gameToAdd =
> > +                gamesToAdd.Where(g => g.Thumbnail != null)
>
> If I'm reading this correctly, it will take the first game that shows up
> in the results ordered by highest BGG rating and sets this game? Doesn't
> this have a very high probability of setting stuff to the wrong game? For
> example, if I just created a game and called it "cards" or "family game"
> wouldn't it switch their game to an arbitrary game with these words? I just
> want to make sure we don't have a really confusing result for people who
> have deliberately not selected a BGG game. Thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NemeStats/NemeStats/pull/377/files#r55940496>.
>